st2actions/st2actions/policies/concurrency.py

```diff
@@ -69,14 +69,22 @@ def _apply_before(self, target):
         publish = (status == action_constants.LIVEACTION_STATUS_CANCELING)
         target = action_service.update_status(target, status, publish=publish)
 
+        LOG.debug('Publishing: %s', publish)
+        LOG.debug('Status: %s', status)
+
         return target
 
     def apply_before(self, target):
         target = super(ConcurrencyApplicator, self).apply_before(target=target)
 
-        # Exit if target not in schedulable state.
-        if target.status != action_constants.LIVEACTION_STATUS_REQUESTED:
-            LOG.debug('The live action is not schedulable therefore the policy '
+        valid_states = [
+            action_constants.LIVEACTION_STATUS_REQUESTED,
+            action_constants.LIVEACTION_STATUS_DELAYED,
+            action_constants.LIVEACTION_STATUS_POLICY_DELAYED,
+        ]
```

Comment:
Policy delay happens when a policy delays it. Delay happens if there's an explicit delay. This is a function of how policies are applied, manipulating the liveaction directly instead of returning the desired status.

Reply:
I mean add more comments in the code.  Provide description in the comments how/source of the LiveAction reaching the scheduler in specific status.